src/Core.UnitTests/CFamily/EffectiveRulesConfigCalculatorTests.cs

```diff
@@ -32,20 +34,23 @@ public class EffectiveRulesConfigCalculatorTests
         [TestMethod]
         [DataRow(true /* custom settings are null */)]
         [DataRow(false /* custom settings are not null, but emtpy */)]
-        public void GetConfig_NullCustomSettings_DefaultConfigReturned(bool sourceCustomSettingsAreNull)
+        public void GetConfig_NoCustomSettings_NoError_DefaultsReturned(bool customSettingsAreNull)
         {
             // Arrange
             var testLogger = new TestLogger();
             var testSubject = new EffectiveRulesConfigCalculator(testLogger);
 
-            var defaultRulesConfig = CreateMockConfig("language1");
-            var sourcesSettings = sourceCustomSettingsAreNull ? null : new RulesSettings();
+            var defaultRulesConfig = CreateWellKnownRulesConfig("language1");
+            var sourcesSettings = customSettingsAreNull ? null : new RulesSettings();
 
             // Act
             var result = testSubject.GetEffectiveRulesConfig("language1", defaultRulesConfig, sourcesSettings);
 
-            // Assert - optimisation - expecting the same object instance
-            result.Should().BeSameAs(defaultRulesConfig);
+            // Assert
+            result.Should().NotBeSameAs(defaultRulesConfig);
```

Comment:
Does it matter that it's not the same instance ref?
I think it's a bit confusing. The important assertion is the rule keys below, so I'd keep that one and remove this one.

Reply:
Well, it's part of checking that the code goes through a single point before returning. Previously the optimisation meant the original object would be returned, but that definitely isn't what we want now. Still, it's not a strong check so I'll remove it if it's confusing.